clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -791,10 +791,24 @@ static void VisitField(CXXRecordDecl *Owner, RangeTy &&Item, QualType ItemTy,
   else if (ItemTy->isStructureOrClassType())
     VisitAccessorWrapper(Owner, Item, ItemTy->getAsCXXRecordDecl(),
                          handlers...);
+    // FIXME Enable this when structs are replaced by their fields
+#define STRUCTS_DECOMPOSED 0
```

Comment:
The macro here is formed pretty wackily.  I'd hope this doesn't get into the final patch.

Reply:
This macro is only to illustrate what code needs to remain/removed when this patch is finally merged with PR #1877 changes. It is removed in PR#1877

Reply:
Note that this is an interim checkin, until structs are decomposed.

This macro is simply to explain to the reviewers what the code does now, and how it will be changed when full struct support is enabled.

Along with the struct decomposition changes you are free to replace any parts of this that you consider fairly mediocre.